zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/RestErrorMapper.java

```diff
@@ -327,4 +319,53 @@ public static ResponseEntity<Object> mapDocumentHandlingExceptionToResponse(
       final DocumentException e) {
     return mapProblemToResponse(mapDocumentHandlingExceptionToProblem(e));
   }
+
+  public static ProblemDetail mapCamundaSearchExceptionToProblem(
+      CamundaSearchException cse, final Function<BrokerRejection, ProblemDetail> rejectionMapper) {
+    final CamundaSearchException.Reason reason =
+        (cse.getReason() == null) ? CamundaSearchException.Reason.UNKNOWN : cse.getReason();
+    final Throwable cause = (cse.getCause() == null) ? cse : cse.getCause();
+    final String name = cause.getClass().getName();
```

Comment:
🔧 We should only need the name in case of an UNKNOWN reason. Otherwise, the reason should be the `title` in the problem detail.